sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -455,21 +506,26 @@ public void process(
     @FinishBundle
     public void finishBundle(FinishBundleContext context) throws Exception {
       flushAll();
-      for (DestinationState state : destinations.values()) {
-        if (!useDefaultStream) {
+      if (!useDefaultStream) {
```

Comment:
Why this change? Won't this prevent us from unpinning the client?

Reply:
it will, but those clients can be reused across bundle executions.

Reply:
the clients can be reused regardless - unpinning won't close the client unless close is also called (i.e. if we hit the cache idle timeout)